src/app/modules/submission/email-submission/email-submission.service.ts

```diff
@@ -1,37 +1,56 @@
 import crypto from 'crypto'
 import stringify from 'json-stringify-deterministic'
 import { flatten, sumBy } from 'lodash'
+import mongoose from 'mongoose'
 import { errAsync, okAsync, ResultAsync } from 'neverthrow'
 
 import { sessionSecret } from '../../../../config/config'
 import { createLoggerWithLabel } from '../../../../config/logger'
-import { FieldResponse } from '../../../../types'
+import {
+  FieldResponse,
+  IEmailFormSchema,
+  IEmailSubmissionSchema,
+  SubmissionType,
+} from '../../../../types'
+import { getEmailSubmissionModel } from '../../../models/submission.server.model'
 import {
   isProcessedCheckboxResponse,
   isProcessedTableResponse,
 } from '../../../utils/field-validation/field-validation.guards'
+import { DatabaseError } from '../../core/core.errors'
 import { ProcessedFieldResponse } from '../submission.types'
 
+import {
+  DIGEST_TYPE,
+  HASH_ITERATIONS,
+  KEY_LENGTH,
+  SALT_LENGTH,
+} from './email-submission.constants'
```

Comment:
it might be better to inline these constants in this file if the constants aren't used anywhere else. do we use the same scheme for say, myinfo hash?

Reply:
it's used in tests!